src/core/components/files.js

```diff
@@ -137,10 +137,19 @@ module.exports = function files (self) {
       pull.collect((err, files) => {
         if (err) { d.end(err) }
         if (!files || !files.length) {
-          return d.end(new Error('No such file'))
+          return d.abort(new Error('No such file'))
         }
 
-        const content = files[files.length - 1].content
+        if (files.length > 1) {
+          files = files.filter((file) => file.path === ipfsPath)
```

Comment:
Here `file.path` is the path from the root e.g. `tour` and `tour/0.0-intro`, and ipfsPath is the full path e.g. `/ipfs/QmfGBRT6BbWJd7yUc2uYdaUZJBbnEFvTqehPFoSMQ6wgdr/tour`.

This filter is removing all the files from the array and so `files[0]` below is `undefined`

<img width="1392" alt="screen shot 2017-12-14 at 12 13 48" src="https://user-images.githubusercontent.com/152863/33991856-4dc7b12e-e0c8-11e7-819c-f92c69ed7f59.png">

Reply:
Thanks @alanshaw !
#1148 should fix this.

Reply:
Theres's still an issue here, where "tour" dir is being filtered out prematurely because the output path doesn't match the input. Digging in...

Reply:
... which should be fixed f6c15c6a9d9a0847e584910879f2d0402524cc33